### PR TITLE
FES Robots and Indexing

### LINKED ISSUE #8585: FES - Add section for robots.txt for Gatsby and Next
Re: <Add Doc title and/or link>

Priority: Low/Medium/High (choose one, remove the other options)
Low

## Issue Description

* To get some parity with CMS sites docs, add a doc explaining how to set `robots.txt` with Gatsby and Next.js
* Possibly link to this page from the CMS sites doc on `robots.txt` as related?

### How will this impact users?

* Users will be confident in setting `robots.txt` on their Front-End Sites

### Context

* A member of the Pantheon slack was unsure if the current docs on Bots and Indexing applied to FES. They do not and we should document how to update `robots.txt` for our officially supported frameworks

## Suggested Resolution

* Document the following for Front-End Sites:

Gatsby plugin for setting `robots.txt`: https://www.gatsbyjs.com/plugins/gatsby-plugin-robots-txt/

For Next.js, users can add a `robots.txt` file to the `public` folder where static assets go: https://nextjs.org/learn/seo/crawling-and-indexing/robots-txt

### PR DESCRIPTION
**Pull request by whitneymeredith**
Closes #8585 

## Summary

**[Robots and Indexing](https://docs.pantheon.io/guides/decoupled/wp-nextjs-frontend-starters/robots-indexing)** - Adds new pages to document how users of Next.js (WP and Drupal) and Gatsby (WP only) frameworks can use a robots.txt file on their Front-End Sites.

## Effect

The following changes are already committed:

*
*

## Remaining Work and Prerequisites

The following changes still need to be completed:

- [ ] List any outstanding work here

### Dependencies and Timing

- [ ] Other prerequisites that must be completed before merging this PR

**Release**:
- [ ] When ready
- [ ] After date: $DATE

--------------------------------------------------

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/old-path/` => `/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

File: source/content/partials/decoupled-nextjs-robots.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+## Add a robots.txt file to a Next.js Project
+
+You can add a `robots.txt` file to your Front-End Sites project using the Next.js static file serving feature. The steps below are intended as a guide to help you get started. Refer to [Next.js Crawling and Indexing documentation](https://nextjs.org/learn/seo/crawling-and-indexing/robots-txt) for more information.
+
+1. Navigate to your root directory and then open the `public` folder.
+
+1. Create a file named `robots.txt` and specify which URLs you want to block or allow crawler access to. For example:
+
+    ```bash{promptUser: user}
```
Comment (CobyPear) [inline source/content/partials/decoupled-nextjs-robots.md:23]: Can this be a regular text file? The bash prompt renders the `$` on each line of the output which is a bit confusing

File: source/content/guides/decoupled/wp-gatsby-frontend-starters/05-robots-indexing.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    **npm:**
+
+    ```bash{promptUser: user}
+    npm install --save gatsby-plugin-robots-txt
+    ```
+
+1. Open your `gatsby-config.js` file and add the URLs you want to block or allow crawler access to.
+
+    ```bash{promptUser: user}
```
Comment (CobyPear) [inline source/content/guides/decoupled/wp-gatsby-frontend-starters/05-robots-indexing.md:46]:
```suggestion
    ```javascript
```
This one can be `javascript` if that is supported.

File: source/content/guides/decoupled/wp-gatsby-frontend-starters/05-robots-indexing.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  siteMetadata: {
+    siteUrl: 'https://www.example.com'
+  },
+  plugins: ['gatsby-plugin-robots-txt']
+    };
+    ```
+
+    **Example configuration:**
+
+     ```bash{promptUser: user}
```
Comment (CobyPear) [inline source/content/guides/decoupled/wp-gatsby-frontend-starters/05-robots-indexing.md:59]: Same here
```suggestion
     ```javascript
```

File: source/content/guides/decoupled/wp-gatsby-frontend-starters/05-robots-indexing.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+        //gatsby-config.js
+        module.exports = {
+    plugins: [
+        {
+        resolve: 'gatsby-plugin-robots-txt',
+        options: {
+            host: 'https://www.mygatsbysite.com',
+            sitemap: 'https://www.mygatsbysite.com/sitemap.xml',
```
Comment (CobyPear) [inline source/content/guides/decoupled/wp-gatsby-frontend-starters/05-robots-indexing.md:68]:
```suggestion
            host: 'https://www.mygatsbysite.example',
            sitemap: 'https://www.mygatsbysite.example/sitemap.xml',
```
I learned about this the other day: use `.example` or `example.com` as the domain for documenting examples
https://datatracker.ietf.org/doc/html/rfc2606#section-2

**Review by CobyPear — CHANGES_REQUESTED** [review]
Looks great overall– just a few picky things in the comments.

**Review by CobyPear — APPROVED** [review]

**Review by wordsmither — APPROVED** [review]
lgtm

**Pull request merged**